### PR TITLE
Change Image Name to Use Artifact Registry

### PR DESCRIPTION
**Pull request by sealesj**
In order to generate provenance and increase SLSA level, images are being routed to Artifact Registry instead of Container Registry. The transition has been tested for both autosubmit and app-dart. It is also necessary for the flutter image build prerequisite to be pushed to Artifact Registry.

 As part of this, the build configurations for app-dart and flutter image build need to be updated to point to the correct Artifact Registry location.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read the [Flutter Style Guide] _recently_, and have followed its advice.
- [x] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Review by drewroengoogle — APPROVED** [review]
LGTM! Additional context: this change is based on the redirection mappings referenced here: https://cloud.google.com/artifact-registry/docs/transition/setup-gcr-repo#mapping

File: app_dart/cloudbuild_app_dart.yaml [hunk 1]
```diff
@@ -6,7 +6,7 @@
 
 steps:
   # Build dashboard.
-  - name: gcr.io/$PROJECT_ID/flutter
+  - name: us-docker.pkg.dev/$PROJECT_ID/gcr.io/$PROJECT_ID/flutter
```
Comment (keyonghan) [inline app_dart/cloudbuild_app_dart.yaml:9]: Shall this be: `us-docker.pkg.dev/$PROJECT_ID/gcr.io/flutter`

Comment (sealesj) [reply app_dart/cloudbuild_app_dart.yaml:9]: Yes, that would be simpler. We will need to also change the `flutter_image_build.yaml` to upload with that name, but that change will happen after this one regardless.

Comment (sealesj) [reply app_dart/cloudbuild_app_dart.yaml:9]: We copied over the image with the simplified name and are running a test to pull it down with that path

File: app_dart/cloudbuild_app_dart.yaml [hunk 2]
```diff
@@ -6,7 +6,7 @@
 
 steps:
   # Build dashboard.
-  - name: gcr.io/$PROJECT_ID/flutter
+  - name: us-docker.pkg.dev/$PROJECT_ID/gcr.io/flutter
```
Comment (drewroengoogle) [inline app_dart/cloudbuild_app_dart.yaml:9]:
```suggestion
  - name: us-docker.pkg.dev/$PROJECT_ID/flutter/flutter
```

**Comment by sealesj** [comment]
@keyonghan We still need an enclosing folder, hence the flutter/flutter, but we can simplify to use the Project ID variable only once and also remove the gcr.io

**Comment by sealesj** [comment]
Updated with flutter image build location change

**Review by keyonghan — APPROVED** [review]
LGTM

**Pull request merged**